frontend/catalyst/compiler.py

```diff
@@ -159,6 +159,19 @@ def run_writing_command(command: List[str], compile_options: Optional[CompileOpt
     subprocess.run(command, check=True)
 
 
+TAPE_SPLITTING_PASS = (
+    # We clump multiple tapes into a single function and split them
+    # in mlir with a pass (frontend/jax_tracer.py).
+    # Therefore before the splitting the quantum mlir is "illegal",
+    # as each function will have multiple devices.
+    # Thus, the split must be the very first pass before anything else
+    # happens.
```

Comment:
Is this an actual restriction or could we easily perform the splitting as the first pass in the quantum compilation pipeline? 🤔 (not that we necessarily should, but I'm curious)

Reply:
I found out that split tapes must occur before `cse` in `HLOLoweringPass`. I'm not sure why.